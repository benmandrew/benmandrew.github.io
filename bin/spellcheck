#!/bin/sh

jq_program='[.[]
  | {file: .history[0], messages: [.messages[] | {message: .message | ltrimstr("`") | rtrimstr("; did you mean ?") | rtrimstr("` is misspelt"), line, column}]}
  | select(.messages | length > 0)]'

python_program='import json
with open("filtered.json", "r") as f:
  data = json.load(f)
  for file in data:
    print(file["file"])
    for m in file["messages"]:
      loc = "{}:{}".format(m["line"], m["column"])
      print("  {:7} - {}".format(loc, m["message"]))
    print()'

spellchecker -q --no-suggestions --language en-GB _posts/* --reports all.json --dictionaries bin/spellcheck.dictionary 1>/dev/null

jq $jq_program all.json > filtered.json

python3 -c $python_program > spellcheck.actual

diff bin/spellcheck.expected spellcheck.actual
