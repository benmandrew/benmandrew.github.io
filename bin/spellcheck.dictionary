pixelated
Bézier
Bezier
bézier
bezier
wikipedia
pooper
lerp
gifs
renderer
flipbook
Voronoi
FFMPEG
ffmpeg
RGB
PNG
framerate
roguelikes
roguelike
customisable
repurposing
octant
Github
raytracing
3D
cryo-pod
shaders
WASD
shader
mipmap
mipmaps
blits
blitting
blit
Shader
Shaders
laggy
finangling
greyscale
Perlin
Sobel
GPUs
Javascript
Renderer
rasterisation
repo
normals
barycentric
rasterised
projective
renormalise
rasterising
ourself
rasterising
unnormalised
2D
janky
leaderboard
speedrunners
VODs
VOD
performant
naïve
bottlenecked
cacheline
CPUs
cachelines
subarrays
subarray
mergesort
malloc
Mergesort
OCaml
Microcontroller
microcontroller
Tarides
bytecode
keypress
samaritan
loopback
keypresses
topologies
codeword
ocaml
codewords
recurse
recursing
Parallelising
Hoare
postcondition
invariants
prover
composable
Coq
Automata
automata
Recursing
toruses
MacOS
