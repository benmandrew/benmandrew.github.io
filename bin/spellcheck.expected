_posts/2017-08-18-trench-warfare.markdown
  10:1    - fullwidth

_posts/2018-04-21-writing-a-bezier-curve-plotter.markdown
  14:41   - Pygame
  160:60  - github
  161:105 - Unity3D
  170:136 - 4D

_posts/2018-04-30-3d-bezier-curves.markdown
  10:1    - fullwidth

_posts/2018-05-02-connect-4-ai.markdown
  10:1    - fullwidth

_posts/2018-05-30-revenant.markdown
  10:1    - fullwidth
  44:10   - powerups
  45:17   - powerup

_posts/2018-06-23-lux-project.markdown
  14:47   - UI
  14:150  - tkinter

_posts/2019-05-19-graphics-rejects.markdown
  10:1    - fullwidth

_posts/2019-09-28-creating-videos-from-scratch-with-ffmpeg.markdown
  14:104  - WMV
  14:119  - MP4
  54:118  - NumPy
  60:158  - PIL
  60:211  - a-ok
  60:230  - NumPy
  88:268  - Shlex

_posts/2019-10-01-copper-logic.markdown
  10:1    - fullwidth

_posts/2020-01-06-field-of-view-in-a-tile-based-world.markdown
  33:111  - NPCs
  44:19   - Bresenham's
  58:6    - Bresenham's
  117:282 - enum

_posts/2020-02-06-ray-tracing-with-bounding-volume-hierarchies.markdown
  12:90   - mn
  103:24  - BVH
  105:183 - UVs
  105:216 - BVH
  107:26  - BVH
  107:336 - BV
  133:2   - Welford's
  135:71  - BV
  137:287 - BVs
  163:27  - BVH
  163:349 - BVH
  165:111 - quicksort
  165:369 - quicksort
  165:414 - quicksort's
  184:26  - BVH
  186:53  - BV
  204:36  - BVH

_posts/2020-04-21-cryogen.markdown
  10:1    - fullwidth
  30:22   - Ludum
  32:76   - Ludum

_posts/2020-09-28-custom-post-processing-effects-in-unity.markdown
  21:7    - RenderTexture
  23:3    - RenderTexture
  26:7    - Blitting
  28:1    - Blitting
  28:46   - RenderTexture
  28:206  - RTs
  31:27   - RenderTexture
  39:55   - GameObject
  115:40  - GameObject
  115:125 - FixedUpdate
  169:95  - XY

_posts/2020-10-06-cosmic-taxi.markdown
  10:1    - fullwidth
  29:34   - CTRL
  32:22   - Ludum
  34:80   - Ludum

_posts/2021-01-27-forward-and-deferred-rendering.markdown
  10:1    - fullwidth
  36:4    - GTA
  42:35   - fdr
  43:25   - Siedler

_posts/2021-02-26-js-table-renderer.markdown
  29:20   - Use `an` before `HTML`, not `a`
  14:51   - JS
  33:154  - F12
  33:231  - JS
  40:23   - 3x3
  40:67   - 4x4

_posts/2021-04-26-goblin-heist.markdown
  12:38   - Ludum
  24:65   - Ludum
  26:64   - Tobugis
  27:65   - ColeSlawski

_posts/2021-06-30-coolsort-or-cache-access-pattern-optimisation.markdown
  3:14    - CoolSort
  27:75   - L1
  42:192  - CoolSort
  44:187  - cacheline-aligned
  54:29   - mergesort's
  63:77   - nlogn
  111:27  - sizeof
  113:2   - isPrimaryArr
  114:15  - INTS_IN_CACHELINE
  121:29  - isPrimaryArr
  121:45  - isPrimaryArr
  121:66  - isPrimaryArr
  142:132 - O3
  142:193 - I5-5300U
  146:5   - logX-Y
  146:136 - Coolsort
  154:31  - Coolsort
  154:59  - Mergesort's
  158:17  - Coolsort
  167:82  - Von
  167:86  - Neumann
  168:145 - L1

_posts/2021-08-28-porting-ocaml-to-nrf52.markdown
  3:35    - nRF52
  10:1    - fullwidth
  27:12   - OMicroB

_posts/2022-02-25-unix-select-syscall.markdown
  123:4   - DTLSR

_posts/2022-05-16-dtlsr-project.markdown
  12:177  - LSR
  12:185  - DTLSR
  12:365  - OSPF
  16:1    - LSR
  16:9    - DTLSR
  16:147  - DTLSR
  16:156  - LSR
  21:3    - DTLSR

_posts/2022-07-12-ocaml-tricks-for-long-computations.markdown
  14:21   - LZW
  58:128  - cons-ing
  127:7   - LZW

_posts/2022-11-03-parallelising-a-software-ray-tracer.markdown
  10:1    - fullwidth
  19:151  - domainslib

_posts/2023-02-16-dummys-guide-to-logic-and-proof.markdown
  18:535  - rightarrow
  18:601  - rightarrow
  20:44   - rightarrow
  20:268  - modus
  20:274  - ponens
  24:2    - AxiomC
  25:2    - AxiomC
  25:13   - rightarrow
  26:2    - BinaryInfC
  30:158  - rightarrow
  30:230  - rightarrow
  30:510  - rightarrow
  34:2    - AxiomC
  35:2    - AxiomC
  35:13   - rightarrow
  36:2    - BinaryInfC
  37:2    - AxiomC
  37:13   - rightarrow
  38:2    - BinaryInfC
  44:46   - modus
  44:52   - ponens
  48:2    - AxiomC
  49:2    - AxiomC
  50:2    - BinaryInfC
  54:99   - vee
  58:2    - AxiomC
  59:2    - UnaryInfC
  59:16   - vee
  63:2    - AxiomC
  64:2    - UnaryInfC
  64:16   - vee
  68:60   - rightarrow
  68:89   - vee
  68:245  - vee
  70:188  - rightarrow
  70:266  - rightarrow
  74:2    - AxiomC
  75:2    - AxiomC
  75:12   - rightarrow
  76:2    - BinaryInfC
  77:2    - AxiomC
  78:2    - AxiomC
  78:12   - rightarrow
  79:2    - BinaryInfC
  80:2    - BinaryInfC
  84:338  - rightarrow
  88:439  - rightarrow
  141:72  - rightarrow
  143:182 - rightarrow
  145:19  - rightarrow
  145:205 - vee
  167:45  - rightarrow
  167:60  - vee
  167:249 - rightarrow
  167:285 - vee
  171:2   - AxiomC
  171:13  - rightarrow
  172:2   - UnaryInfC
  172:16  - rightarrow
  172:31  - vee
  176:46  - vee
  176:107 - vee
  178:25  - rightarrow
  178:40  - vee
  178:151 - rightarrow
  178:166 - vee

_posts/2023-06-01-cavalry.markdown
  29:4    - frac
  31:441  - SMT
  35:6    - rightarrow
  35:17   - wp
  37:9    - wp
  37:129  - wp
  37:188  - wp
  39:28   - rightarrow
  39:39   - wp
  41:53   - sp
  41:150  - sp
  41:321  - wp
  45:501  - pre-
  47:4    - frac
  47:40   - texttt
  47:61   - texttt
  47:79   - texttt
  68:164  - frac
  125:84  - MPRI

_posts/2023-07-01-pctl.markdown
  3:48    - PCTL
  33:1    - PCTL
  33:135  - PCTL
  36:11   - leq
  36:20   - geq
  48:12   - leq
  48:22   - geq
  48:34   - texttt
  62:2    - texttt
  62:18   - leq
  62:27   - geq
  62:37   - texttt
  62:55   - texttt
  69:39   - texttt
  69:111  - texttt
  69:265  - texttt
  69:313  - texttt
  69:352  - texttt
  69:370  - texttt
  69:449  - texttt
  69:468  - texttt
  73:328  - texttt
  73:346  - texttt
  73:377  - texttt
  75:198  - texttt
  75:216  - texttt
  75:364  - texttt
  77:231  - texttt
  77:247  - leq
  77:256  - geq
  77:266  - texttt
  77:284  - texttt
  83:15   - texttt
  84:10   - texttt
  85:5    - texttt
  86:18   - texttt
  87:9    - texttt
  87:23   - texttt
  87:41   - texttt
  87:55   - texttt
  87:71   - leq
  87:80   - geq
  87:90   - texttt
  87:108  - texttt
  93:393  - leq
  93:402  - geq
  98:20   - notin
  105:116 - geq
  107:147 - texttt
  107:163 - leq
  107:172 - geq
  107:182 - texttt
  107:200 - texttt
  121:90  - texttt
  121:106 - leq
  121:115 - geq
  121:125 - texttt
  121:143 - texttt
  121:168 - leq
  126:3   - PCTL
  129:8   - Hansson
  129:20  - Bengt
  129:26  - Jonsson
  130:3   - Yunjeong

_posts/2024-03-13-rendering-geometric-patterns.markdown
  10:1    - fullwidth
  19:199  - Critchlow
  50:16   - pdf
  50:110  - odp

_posts/2024-03-16-bdd.markdown
  19:47   - BDDs
  19:129  - BDDs
  25:82   - sub-BDDs
  25:118  - BDD
  34:13   - Rightarrow
  36:27   - BDD
  36:45   - Rightarrow
  36:114  - BDD
  36:155  - BDD
  42:43   - BDD
  42:59   - sub-BDD
  46:10   - BDD
  46:72   - BDD
  52:196  - sub-BDD
  52:265  - Rightarrow
  52:304  - Rightarrow
  52:422  - BDD
  58:68   - sub-BDD
  68:13   - sub-BDDs
  68:85   - BDDs
  98:34   - BDD
  112:81  - BDD
  114:255 - BDD
  121:85  - Rightarrow
  123:243 - BDDs
  123:274 - oplus
  125:14  - Rightarrow
  125:40  - iff
  125:58  - oplus
  127:25  - BDD
  127:61  - Rightarrow
  127:100 - BDD
  127:123 - oplus
  127:144 - iff
  127:180 - BDD
  127:225 - BDD
  134:41  - iff
  136:82  - sub-BDDs
  136:131 - BDD
  140:9   - sub-BDDs
  140:66  - BDD
  147:53  - iff
  147:101 - sub-BDD
  161:9   - sub-BDDs
  161:66  - BDD
  172:9   - sub-BDDs
  179:54  - iff
  179:94  - BDD
  179:185 - BDD
  189:9   - sub-BDDs
  196:23  - iff
  204:32  - sub-BDDs
  204:93  - BDD
  204:112 - Rightarrow
  204:138 - iff
  204:156 - oplus

_posts/2024-04-09-terminal-renderer.markdown
  18:47   - Notty
  20:92   - Asciinema
  37:172  - iTerm2

